zilliqa/src/api/eth.rs

```diff
@@ -816,6 +766,377 @@ fn get_uncle(_: Params, _: &Arc<Mutex<Node>>) -> Result<Option<String>> {
     Ok(None)
 }
 
+#[derive(Deserialize, Default)]
+#[serde(default, rename_all = "camelCase")]
+struct NewFilterParams {
```

Comment:
Should we consider using https://docs.rs/alloy/latest/alloy/rpc/types/eth/struct.Filter.html instead?

I also node that it supports a block hash as a parameter which I guess is part of the spec?

Reply:
The block hash isn't mention in the spec (https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_newfilter) and that filter struct is used for a different api endpoint (https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getfilterlogs) that does include block hash as a parameter

Reply:
But it might just be better to use that struct anyway

Reply:
Having looked at it again I don't think it makes sense to have the block hash parameter, as that block would always be in the past and therefore this endpoint would not be suitable. Please correct me if I am wrong.